### ISSUE TITLE
Add rule: prefer-explicit-return-type

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#4]
## Problem

LLMs often omit return type annotations on exported functions, making API contracts unclear.

## Rule

Enforce explicit return types on exported functions.

### ❌ Bad
```ts
export function fetchUser(id: string) {
  return api.get(`/users/${id}`);
}
```

### ✅ Good
```ts
export function fetchUser(id: string): Promise<User> {
  return api.get(`/users/${id}`);
}
```

## Why it matters

- Makes API contracts explicit
- Better IDE autocomplete
- Self-documenting code
